### PR TITLE
Support BigQuery DATETIME and TIME types

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java [hunk 1]
```diff
@@ -95,7 +99,9 @@ public void testConvertGenericRecordToTableRow() throws Exception {
       record.put("birthday", 5L);
       record.put("flighted", Boolean.TRUE);
       record.put("sound", soundByteBuffer);
-      record.put("anniversary", new Utf8("2000-01-01"));
+      record.put("anniversary_date", new Utf8("2000-01-01"));
+      record.put("anniversary_datetime", new String("2000-01-01 00:00:00.000005"));
```
Comment (lukecwik) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java]: Did you sync with BQ about the inconsistencies with the date time types you were having where input and output differed.

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryTableRowIteratorTest.java [hunk 2]
```diff
@@ -168,11 +170,13 @@ public void testReadFromQuery() throws IOException, InterruptedException {
...
-    when(mockTabledataList.execute())
-        .thenReturn(rawDataList(rawRow("Arthur", 42, photoBytesEncoded, "2000-01-01")));
+    when(mockTabledataList.execute()).thenReturn(
+        rawDataList(rawRow("Arthur", 42, photoBytesEncoded,
+            "2000-01-01", "2000-01-01 00:00:00.000005", "00:00:00.000005")));
 
     // Run query and verify
-    String query = "SELECT name, count, photo, anniversary from table";
+    String query = "SELECT name, count, photo, anniversary_date, "
+        + "anniversary_datetime, anniversary_time  from table";
```
Comment (lukecwik) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryTableRowIteratorTest.java]: remove extraneous space after anniversary_time

**Comment by lukecwik** [comment]
R: @lukecwik

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryTableRowIteratorTest.java (hunk 2 above)
Comment (peihe) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryTableRowIteratorTest.java]: done

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java (hunk 1 above)
Comment (peihe) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java]: yes, I filed a bug. And, our code should just work once the bug is fixed on BQ side.

**Comment by peihe** [comment]
tests passed
PTAL

**Comment by lukecwik** [comment]
LGTM, merging

**Pull request closed without merging**